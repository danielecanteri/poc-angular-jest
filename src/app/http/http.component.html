<p *ngIf="!film">
  loading
</p>
<p *ngIf="film">
  {{film.title}}
</p>
