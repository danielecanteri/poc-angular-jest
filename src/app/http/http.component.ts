import { Component, OnInit } from '@angular/core';
import { Http } from "@angular/http";

@Component({
  selector: 'app-http',
  templateUrl: './http.component.html',
  styleUrls: ['./http.component.css']
})
export class HttpComponent implements OnInit {
  film: any;

  constructor(
    private http: Http
  ) { }

  ngOnInit() {
    this.http.get('http://swapi.co/api/films/1/').subscribe(
      (response) => { this.film = response.json() }
    );
  }

}
