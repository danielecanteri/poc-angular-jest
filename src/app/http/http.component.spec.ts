import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { HttpComponent } from './http.component';
import { Http } from "@angular/http";
import { Observable } from "rxjs/Observable";
import 'rxjs/add/observable/of';
import { Subject } from "rxjs/Subject";

describe('HttpComponent', () => {
  let component: HttpComponent;
  let fixture: ComponentFixture<HttpComponent>;

  let response = new Subject();

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [HttpComponent],
      providers: [
        {
          provide: Http,
          useValue: {
            get: function () { return response; }
          }
        }
      ]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HttpComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should show when still loading', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('shows result', () => {
    response.next({
      json: function () {
        return {
          title: 'A new Hope'
        }
      }
    })
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

});
