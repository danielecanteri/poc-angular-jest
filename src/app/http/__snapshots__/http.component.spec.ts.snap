// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HttpComponent should show when still loading 1`] = `
<app-http
  http={[Function Object]}
>
  <p>
    loading
  </p>
</app-http>
`;

exports[`HttpComponent shows result 1`] = `
<app-http
  film={[Function Object]}
  http={[Function Object]}
>
  <p>
    A new Hope
  </p>
</app-http>
`;
